src/Compilers/CSharp/Portable/Symbols/MemberSymbolExtensions.cs

```diff
@@ -658,5 +658,16 @@ internal static string GetMemberCallerName(this Symbol member)
                 member.IsExplicitInterfaceImplementation() ? ExplicitInterfaceHelpers.GetMemberNameWithoutInterfaceName(member.Name) :
                 member.Name;
         }
+
+        internal static bool NeedFieldKeywordBinder(this Symbol symbol)
```

Comment:
Rather than call this `NeedFieldKeywordBinder`, I think a better name would be `CanHaveFieldKeywordBackingField` or similar. The use of this method in the binder itself is misleading with the current naming: we check to see if we need a specific binder, then just blindly get the backing field?